MapboxNavigation/NavigationMapView.swift

```diff
@@ -94,22 +100,66 @@ open class NavigationMapView: MGLMapView {
     }
     
     func shape(describing route: Route) -> MGLShape? {
-        guard var coordinates = route.coordinates else {
-            return nil
+        guard let coordinates = route.coordinates else { return nil }
+        
+        let congestionPerLeg = route.legs.flatMap {
+            $0.segmentCongestionLevels
+        }
+        
+        let combinedCongestionLevel = Array(congestionPerLeg.joined()) // Flatten all leg nodes
+        let destination = coordinates.suffix(from: 1)
+        let segment = zip(coordinates, destination).map { [$0.0, $0.1] }
```

Comment:
We wind up zipping together a full segment array even if `congestionPerLeg` is the empty array. This probably happens in the Objective-C example, which doesn’t specify the `congestion` attribute when requesting the route.